graphix/graphsim/basegraphstate.py

```diff
@@ -235,9 +415,15 @@ def equivalent_graph_E2(self, node1, node2):
         ----------
         node1, node2 : int
             connected graph nodes to apply rule E2
+
+        Returns
+        ----------
+        None
         """
-        assert (node1, node2) in list(self.edges) or (node2, node1) in list(self.edges)
-        assert not self.nodes[node1]["loop"] and not self.nodes[node2]["loop"]
+        if (node1, node2) not in list(self.edges) and (node2, node1) not in list(self.edges):
+            raise ValueError("nodes must be connected by an edge")
```

Comment:
From my profiling result, this assertion is the bottleneck of the Pauli measurements procedures. So could you delete this part, or convert it to a more efficient one if possible?